helpers/setup-spark-k8s.py

```diff
@@ -0,0 +1,135 @@
+        else:
+            print('ERROR: No clusters found in kubeconfig file. Please provide a valid kubeconfig file!')
+            sys.exit(-100)
+
+        os.system('\n\necho {} | base64 --decode'.format(certs[int(cluster_id)]))
+
+
+def setup_user(username: str, namespace: str, kubeconfig: str, cluster_name: str) -> None:
+    clusterrolebindingname = username + '-role'
+    clusterroleaccess='view'
+    os.system("{} create serviceaccount --kubeconfig={} --cluster={} {} --namespace={}".format(kubectl_cmd, kubeconfig, cluster_name, username, namespace))
+    os.system("{} create clusterrolebinding --kubeconfig={} --cluster={} {} --clusterrole={}  --serviceaccount={}:{} --namespace={}".format(kubectl_cmd, kubeconfig, cluster_name, clusterrolebindingname, clusterroleaccess, namespace, username, namespace))
+	
+
+def dump_ca_crt(secretname: str, kubeconfig: str, cluster_name: str) -> None:
+    os.system('{} get secret --kubeconfig={} --cluster={} {} -o jsonpath="{{.data.ca\.crt}}" | base64 --decode'.format(kubectl_cmd,  kubeconfig, cluster_name, secretname))
+
+
+def generate_token(serviceaccountname: str, namespace: str, kubeconfig: str, cluster_name: str) -> None:
+    token_expiry_duration='24h' # NOTE: The server may return a token with a longer or shorter lifetime.
+    os.system('{} create token {} --namespace {} --duration {} --kubeconfig={} --cluster={}'.format(kubectl_cmd, serviceaccountname, namespace, token_expiry_duration, kubeconfig, cluster_name))
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--kubeconfig", default='no-kubeconfig-provided', help='Kubernetes configuration file')
+    parser.add_argument("--cluster", default='no-cluster-provided', help='Cluster name to use within the provided kubernetes configuration file')
+    subparsers = parser.add_subparsers(dest='action')
+    subparsers.required = True
+
+    #  subparser for service-account
+    parser_account = subparsers.add_parser('service-account')
+    parser_account.add_argument('username', help='Service account username to be created in kubernetes.')
+    parser_account.add_argument('--namespace', default='default', help='Namespace for the service account to be created in kubernetes. Default is default namespace')
+
+    #  subparser for CA certificate
+    parser_certificate = subparsers.add_parser('get-ca-cert')
+
+    #  subparser for OAuth token
+    parser_token = subparsers.add_parser('get-token')
+    parser_token.add_argument('serviceaccount', help='Service account name for which the Oauth token is to be fetched.')
+    parser_token.add_argument('--namespace', default='default', help='Namespace for the service account for which the Oauth token is to be fetched. Default is default namespace')
+
+    args = parser.parse_args()
+
+    if args.kubeconfig != 'no-kubeconfig-provided':
+        kubeconfig = args.kubeconfig
+    else:
+        kubeconfig = '/home/{}/.kube/config'.format(os.getlogin())
```

Comment:
1. we should also take into consideration `KUBECONFIG` env variable if path not set.
2. default should be `$HOME/.kube/config`, not `/home/{}/.kube/config` if `KUBECONFIG` env variable is not set.